dotCMS/src/main/java/com/dotcms/util/content/json/PopulateContentletAsJSONUtil.java

```diff
@@ -0,0 +1,398 @@
+                }
+
+                try (ResultSet rs = stmt.getResultSet()) {
+
+                    // Now we want to write the found Contentlets into a file for a later processing
+                    var dotConnect = new DotConnect();
+                    dotConnect.fromResultSet(rs);
+
+                    var loadedResults = dotConnect.loadObjectResults();
+                    recordsProcessed += loadedResults.size();
+
+                    if (!loadedResults.isEmpty()) {
+
+                        hasRows = true;
+
+                        var jsonDataArray = Optional.ofNullable(loadedResults)
+                                .map(results ->
+                                        TransformerLocator.createContentletTransformer(results).asList()
+                                )// Transform the results into a list of contentlets
+                                .map(contentletList ->
+                                        contentletList.stream()
+                                                .map(this::toJSON)
+                                                .collect(Collectors.toList())
+                                )// Transform the contentlets into a list of json strings
+                                .orElse(Collections.emptyList());
+
+                        for (var jsonData : jsonDataArray) {
+                            // Write the json representation of the contentlet into the file
+                            fileWriter.write(jsonData);
+                            fileWriter.newLine();
+                        }
+
+                        Logger.debug(this, String.format("Added [%s] records for update to temp file", jsonDataArray.size()));
+
+                    } else {
+                        hasRows = false;
+                    }
+                }
+
+            } while (hasRows);
+
+            // Flush the writer to the file
+            fileWriter.flush();
+
+            // Close the cursor
+            stmt.execute(CLOSE_CURSOR);
+            if (DbConnectionFactory.isMsSql()) {
+                stmt.execute(DEALLOCATE_CURSOR_MSSQL);
+            }
+        } finally {
+            CloseUtils.closeQuietly(fileWriter);
+        }
+
+        Logger.info(this, "-- Records found to process: " + recordsProcessed);
+    }
+
+    /**
+     * This method processes a file that contains all the contentlets that need to be updated with the contentlet_as_json
+     *
+     * @param taskDataFile
+     * @throws IOException
+     */
+    @WrapInTransaction
+    private void processFile(final File taskDataFile) throws IOException {
+
+        Logger.info(this, "Updating records with missing Contentlet as JSON");
+
+        final MutableInt totalUpdateAffected = new MutableInt(0);
+
+        try (final Stream<String> streamLines = Files.lines(taskDataFile.toPath())) {
+
+            streamLines.forEachOrdered(line -> this.processLine(line, totalUpdateAffected));
+
+            if (!this.paramsUpdate.isEmpty()) {
```

Comment:
I think this value should be local, b/c it is not thread safe

Reply:
Done